src/common/pal/src/macos/pal_string.c

```diff
@@ -0,0 +1,126 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+/* Split a string into multiples pieces, divided by a char "match".
+ * Returns a NULL terminated array on success or NULL on error.
+ */
+char **OS_StrBreak(char match, const char *_str, size_t size)
+{
+    size_t count = 0;
+    size_t i = 0;
+    char *dup_str;
+    char *str;
+    char *tmp_str;
+    char **ret;
+    char *str_ant = NULL;
+    char *aux_str = NULL;
+
+    /* We can't do anything if str is null */
+    if (_str == NULL) {
+        return (NULL);
+    }
+
+    dup_str = strdup(_str);
+
+    if(dup_str == NULL) {

```

Comment:
this file is identical to the linux one, can this be avoided and use the same?

Reply:
Wasn't it possible to extract out those minor differences between the two implementations and compile them according to the platform?

Reply:
Once again, it would break the PAL structure, and it would make the PAL itself more untidy.
Each file corresponds to its platform, and in the case of linux  and clang it is repeated because AppleClang its almost compatible with GCC.

Reply:
How would having a common implementation break the PAL structure? Sorry to insist but maybe I need some more clarification :sweat_smile: There could also be a `unix` folder (linux+mac) or a `common` folder (all platforms, if that makes any sense), for code that is the same in every platform you would find the definitions in one of such folders and compile it for all of them, together with the platform specific ones. Again, sorry if it sounds like nitpicking, but if this continues to grow it could become a maintenance problem if we end up having multiple duplicated definitions.

Reply:
An all platform folder does not make sense (since in that case no PAL would be required).
Some way would be to reference from the mac folder the Linux file (I don't particularly like that option).
If it grows and maintenance becames an issue, probably a better structure could be implement from this one.
For the level of complexity it has today, and the benefit it brings (Windows Agent working as the Linux and macOS agents), I don't this it should be applied such changes right now.

Reply:
Ok let's look at it again when it starts growing.